app.py

```diff
@@ -17,45 +19,73 @@
 db = SQLAlchemy(app)
 
 class Users(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
+	id = db.Column(db.Integer, primary_key=True)
 	username = db.Column(db.Text, unique=True, nullable=False)
 	password = db.Column(db.Text, nullable=False)
 	is_admin = db.Column(db.Boolean, nullable=False)
 	is_banned = db.Column(db.Boolean, nullable=False)
+	is_typing = db.Column(db.Boolean, nullable=False)
 
 
 class Subjects(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
+	id = db.Column(db.Integer, primary_key=True)
 	subject = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Groups(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
+	id = db.Column(db.Integer, primary_key=True)
 	groups = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Posts(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
-	user = db.Column(db.BigInteger, db.ForeignKey('users.id'), nullable=False)
-	subject = db.Column(db.BigInteger, db.ForeignKey('subjects.id'), nullable=False)
+	id = db.Column(db.Integer, primary_key=True)
+	user = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+	subject = db.Column(db.Integer, db.ForeignKey('subjects.id'), nullable=False)
 	body = db.Column(db.Text, nullable=False)
-	group = db.Column(db.BigInteger, db.ForeignKey('groups.id'))
+	group = db.Column(db.Integer, db.ForeignKey('groups.id'))
 	time = db.Column(db.DateTime, nullable=False, default=datetime.now())
 
 
-@app.route('/signup', methods=['POST'])
+
+@app.route("/signup", methods=['POST'])
 def signup():
-	...
+	password: str = request.form['password']
+	username: str = request.form['username']
+	if not username or not password:
+		return make_response(jsonify({'task': 'login', 'status': 'failed', 'reason': 'Username and password are required'}), 400)
```

Comment:
if no username or pw were sent to the endpoint it automaticly responses with bad request 404, should wrap it with try catch block